src/library-aux/scala/Any.scala

```diff
@@ -110,13 +110,29 @@ abstract class Any {
    */
   final def ## : Int = sys.error("##")
 
-  /** Test whether the dynamic type of the receiver object is `T0`.
-   *
-   *  Note that the result of the test is modulo Scala's erasure semantics.
-   *  Therefore the expression `1.isInstanceOf[String]` will return `false`, while the
-   *  expression `List(1).isInstanceOf[List[String]]` will return `true`.
-   *  In the latter example, because the type argument is erased as part of compilation it is
-   *  not possible to check whether the contents of the list are of the specified type.
+  /** Test whether the dynamic type of the receiver object has the same erasure as `T0`.
+   *
+   *  Depending on what `T0` is, the test is done in one of the below ways:
+   *
+   *  - `T0` is a non-parameterized class type, e.g. `BigDecimal`: this method returns `true` if
+   *    the value of the receiver object is a `BigDecimal` or a subtype of `BigDecimal`.
+   *  - `T0` is a parameterized class type, e.g. `List[Int]`: this method returns `true` if
+   *    the value of the receiver object is some `List[X]` for any `X`.
```

Comment:
Except for `Array[X]`... worth mentioning?

Reply:
Maybe? Then it becomes so much we can use an examples section, something like

https://github.com/scala/scala/compare/2.13.x...martijnhoekstra:isInstanceOfArray?expand=1

Making that, I noticed that `1.isInstanceOf[1.0]` is false, but `(1: Any) match  { case _: 1.0 => true }` is true, which is not allowed per https://scala-lang.org/files/archive/spec/2.13/12-the-scala-standard-library.html#root-classes

If we're really quiet, maybe nobody else will notice.